### PR TITLE
Vue

### PR DESCRIPTION
**Pull request by dlongley**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # bedrock-views ChangeLog
 
+### Added
+- Support bundle chunking/dynamic imports.
+
```
Comment (davidlehn) [inline CHANGELOG.md]: Needs more info: adding vue, breaking changes, etc

File: lib/vue-sfc-compiler.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
+const bedrock = require('bedrock');
+const async = require('async');
+const fs = require('fs-extra');
+const path = require('path');
+const webpack = require('webpack');
+const memoryfs = require('memory-fs');
+
+const BedrockError = bedrock.util.BedrockError;
+
+const logger = bedrock.loggers.get('app').child('bedrock-views');
```
Comment (davidlehn) [inline lib/vue-sfc-compiler.js]: Could extract common logger to local logger.js like in other modules.

File: lib/config.js [hunk 3]
```diff
@@ -126,11 +146,11 @@ config.views.less.compile.options = {
   sourceMap: false,
   outputSourceFiles: false
 };
-config.views.less.compile.order = ['bootstrap', 'font-awesome'];
+config.views.less.compile.order = [/*'bootstrap', 'font-awesome'*/];
```
Comment (davidlehn) [inline lib/config.js:149]: may be able to remove this.  similar font-awesome code below.

File: lib/packages.js [hunk 4]
```diff
@@ -138,33 +172,64 @@ api.buildImportAllModule = callback => {
...
+      const filename = split.join('/');
+      return {
+        filename: path.resolve(pkg.path, filename),
+        pkg,
+        externals
+      };
+    }
+  }
+  throw new BedrockError(
+    'Not Found.', 'NotFound', {'public': true, httpStatusCode: 400});
```
Comment (davidlehn) [inline lib/packages.js]: Should be 404?

File: lib/system/plugin-vue-loader.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+ * Single File Vue Component Loader.
+ */
+/* global System, URL, fetch */
+exports.fetch = async (url) => {
+  const {vue} = System.getConfig().meta['*.vue'];
+  const path = new URL(url.name).pathname;
+  const response = await fetch(
+    vue.compiler + '?component=' + encodeURIComponent(path));
+  if(response.status !== 200) {
+    const error = await response.json();
```
Comment (davidlehn) [inline lib/system/plugin-vue-loader.js]: Not sure how comprehensive this non-200 path needs to be.  Could get errors that throw while trying to create the message to throw.

File: lib/vue-sfc-compiler.js [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+      callback();
+    });
+  }
+
+  logger.debug(`Compiling Vue SFC: ${task.componentPath}`);
+  async.auto({
+    compile: callback => compile(task, callback),
+    emptyDir: callback => fs.emptyDir(
+      path.dirname(task.cachedFilename), callback),
+    write: ['compile', (results, callback) => {
```
Comment (davidlehn) [inline lib/vue-sfc-compiler.js]: Should this dependy on emptyDir too?

File: lib/system/plugin-vue-loader.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Single File Vue Component Loader.
+ */
+/* global System, URL, fetch */
```
Comment (mattcollier) [inline lib/system/plugin-vue-loader.js]: these globals could be defined in an .eslintrc file in this subdir that specifies an appropriate environment or globals like this: https://github.com/digitalbazaar/bedrock-ledger-agent/blob/master/test/mocha/.eslintrc#L5

File: lib/vue-sfc-compiler.js [hunk 8]
```diff
@@ -0,0 +1,184 @@
+const bedrock = require('bedrock');
```
Comment (mattcollier) [inline lib/vue-sfc-compiler.js:6]: license, copyright, strict?

File: lib/vue-sfc-compiler.js [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+const async = require('async');
+const fs = require('fs-extra');
+const path = require('path');
+const webpack = require('webpack');
+const memoryfs = require('memory-fs');
+
+const BedrockError = bedrock.util.BedrockError;
```
Comment (mattcollier) [inline lib/vue-sfc-compiler.js]: destructured assignment.

File: lib/vue-sfc-compiler.js [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+    module: {
+      rules: [{
+        test: /\.vue$/,
+        use: {
+          loader: require.resolve('vue-loader'),
+          options: {
+            hotReload: false,
+            cssSourceMap: false,
+            loaders: {
+              // disable babel, unnecessary overhead
```
Comment (mattcollier) [inline lib/vue-sfc-compiler.js]: what is going on here?  remove? fix comments?

File: lib/vue-sfc-compiler.js [hunk 11]
```diff
@@ -0,0 +1,184 @@
...
+    if(err) {
+      return callback(err);
+    }
+    if(stats.compilation.errors.length > 0) {
+      logger.error(`SFC compilation error:\nFile: ${filename}\n` +
+        stats.compilation.errors);
+      return callback(new BedrockError(
+        'Could not compile Vue Single Component File.',
+        'DataError', {
+          'public': true,
```
Comment (mattcollier) [inline lib/vue-sfc-compiler.js]: remove quotes.

File: lib/vue-sfc-compiler.js [hunk 12]
```diff
@@ -0,0 +1,184 @@
...
+    callback(null, {bundle, dependencies});
+  });
+}
+
+function mostRecentMtime(filename, deps, callback) {
+  let mtime = 0;
+  async.each(deps, (dep, callback) => {
+    fs.stat(dep, (err, stat) => {
+      if(err) {
+        if(filename === dep) {
```
Comment (mattcollier) [inline lib/vue-sfc-compiler.js]: is `err` when `filename !== dep` being handled properly?  If so, a comment would be helpful.

**Review by mattcollier — CHANGES_REQUESTED** [review]

**Review by davidlehn — APPROVED** [review]

**Pull request merged**